cmd/snap/cmd_snapshot.go

```diff
@@ -0,0 +1,303 @@
+package main
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/jessevdk/go-flags"
+
+	"github.com/snapcore/snapd/client"
+	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/strutil"
+	"github.com/snapcore/snapd/strutil/quantity"
+)
+
+type timeOpts struct {
+	AbsTime bool `long:"absolute-times"`
+	RelTime bool `long:"relative-times"`
+}
+
+var timeDescs = mixinDescs{
+	"absolute-times": i18n.G("Always display absolute times (in YYYY-MM-DD HH:MM format)."),
+	"relative-times": i18n.G("Always display relative times. If neither absolute nor relative times are requested, relative times are used for up to 30 days, and then absolute times in YYYY-MM-DD format."),
+}
+
+func fmtSize(size int64) string {
+	return quantity.FormatAmount(uint64(size), -1)
+}
+
+func (opt timeOpts) fmtTime(t time.Time) string {
+	if opt.AbsTime && !opt.RelTime {
+		return t.Round(time.Minute).Format("2006-01-02 15:04")
+	}
+	ago := time.Since(t)
+	if (opt.RelTime && !opt.AbsTime) || ago < 30*24*time.Hour {
+		// TRANSLATORS: %s is a (separately translated) compact representation of a duration (e.g. 1h30m)
+		return fmt.Sprintf(i18n.G("%s ago"), quantity.FormatDuration(ago.Seconds()))
+	}
+
+	return t.Round(24 * time.Hour).Format("2006-01-02")
+}
+
+type savedCmd struct {
+	timeOpts
+	Wide       bool       `long:"wide"`
+	ID         snapshotID `long:"id"`
+	Positional struct {
+		Snaps []installedSnapName `positional-arg-name:"<snap>"`
+	} `positional-args:"yes"`
+}
+
+var shortSavedHelp = i18n.G("List currently stored snapshots")
```

Comment:
"Lists", if we want to try to stick to that old attempt of making them uniform.

I've moved the original email from the devel list [to the forum](https://forum.snapcraft.io/t/convention-for-command-help-text/1856) some time ago.

Running "snap help" just now, it looks like it derailed again. :(

Reply:
ouch. Will fix.

Reply:
FWIW I fixed this by changing the forum post =) (and, yes, fiing and making all the rest uniform).